src/vm/eetwain.cpp

```diff
@@ -3805,6 +3806,28 @@ bool UnwindEbpDoubleAlignFrame(
     {
         TADDR baseSP;
 
+#ifdef WIN64EXCEPTIONS
+        // Funclets' frame pointers(EBP) are always restored so they can access to main function's local variables.
+        // Therefore the value of EBP is invalid for unwinder so we should use ESP instead.
+        // If funclet frame layout is changed from CodeGen::genFuncletProlog() and genFuncletEpilog(),
+        // we need to change here accordingly.
+        if (pCodeInfo->IsFunclet())
+        {
+            baseSP = curESP + 8; // padding for 16byte stack alignment allocated in genFuncletProlog()
```

Comment:
> The current Windows x86 can handle this situation fine, with just ESP and EBP. What is different about the Linux x86 funclets that it does not work?

`GetPushedArgSize()` is currently only used in ESP-based frames. With funclets enabled for Linux x86, we're generating EBP-based frames. For normal EBP-based frame unwind, we don't need to worry about pushed arguments because we just use the EBP to pop the frame. However, with funclets, we can't do that because the EBP is pointing to the parent function (the funclet sets this up in the prolog to enable stack local variable access via EBP). Maybe it's possible to still use `GetPushedArgSize()` in an EBP-based frame in the funclet case to unwind ESP, and then find the actual EBP that was pushed/saved in the funclet prolog (as opposed to the parent frame EBP that was established)?